jetty-core/jetty-http2/jetty-http2-server/src/main/java/org/eclipse/jetty/http2/server/internal/HttpStreamOverHTTP2.java

```diff
@@ -85,12 +85,14 @@ public Runnable onRequest(HeadersFrame frame)
         {
             _requestMetaData = (MetaData.Request)frame.getMetaData();
 
-            ComplianceViolation.Listener listener = Server.getComplianceViolationListener(_httpChannel.getConnectionMetaData().getConnector()).initialize();
+            // Grab freshly initialized ComplianceViolation.Listener here, no need to reinitialize.
+            ComplianceViolation.Listener listener = _httpChannel.getComponents().getComplianceViolationListener();
```

Comment:
@sbordet This is the exposed violations listeners in components.   I agree it is kind of funny... but is it really any different to the exposed `HttpConfiguration` (see change below)?

The `HttpConfiguration` is public API, but also has getters for lists of `Customizer`s, which users are not meant to use directly.

What if we moved the compliance listeners to `HttpConfiguration`?  They'd still be able to be associated with a particular connector if desired?  It would also be very much more explicit rather than just adding beans.